pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_reconcile.go

```diff
@@ -64,35 +68,103 @@ func (r *endpointsReconciler) reconcile(ctx context.Context, apiExportEndpointSl
 	}
 	apiExport, err := r.getAPIExport(apiExportPath, apiExportEndpointSlice.Spec.APIExport.Name)
 	if err != nil {
+		reason := apisv1alpha1.InternalErrorReason
 		if errors.IsNotFound(err) {
 			// Don't keep the endpoints if the APIExport has been deleted
 			apiExportEndpointSlice.Status.APIExportEndpoints = nil
+			// No need to try again
+			err = nil
+			reason = apisv1alpha1.APIExportNotFoundReason
+		}
+		conditions.MarkFalse(
+			apiExportEndpointSlice,
+			apisv1alpha1.APIExportValid,
+			reason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"Error getting APIExport %s|%s",
+			apiExportPath,
+			apiExportEndpointSlice.Spec.APIExport.Name,
+		)
+		conditions.MarkFalse(
+			apiExportEndpointSlice,
+			apisv1alpha1.APIExportEndpointSliceURLsReady,
+			apisv1alpha1.ErrorGeneratingURLsReason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"",
+		)
+		return err
+	}
+	conditions.MarkTrue(apiExportEndpointSlice, apisv1alpha1.APIExportValid)
+
+	// Get selector from Partition
+	var selector labels.Selector
+	if apiExportEndpointSlice.Spec.Partition != "" {
+		partition, err := r.getPartition(logicalcluster.From(apiExportEndpointSlice), apiExportEndpointSlice.Spec.Partition)
+		rtn := err
+		if err != nil {
+			reason := apisv1alpha1.InternalErrorReason
+			if errors.IsNotFound(err) {
+				// Don't keep the endpoints if the Partition has been deleted
+				// and is still referenced
+				apiExportEndpointSlice.Status.APIExportEndpoints = nil
+				// No need to try again
+				rtn = nil
+				reason = apisv1alpha1.PartitionInvalidReferenceReason
+			}
 			conditions.MarkFalse(
 				apiExportEndpointSlice,
-				apisv1alpha1.APIExportValid,
-				apisv1alpha1.APIExportNotFoundReason,
+				apisv1alpha1.PartitionValid,
+				reason,
 				conditionsv1alpha1.ConditionSeverityError,
-				"APIExport %s|%s not found",
-				apiExportPath,
-				apiExportEndpointSlice.Spec.APIExport.Name,
+				err.Error(),
 			)
-			return nil
-		} else {
 			conditions.MarkFalse(
 				apiExportEndpointSlice,
-				apisv1alpha1.APIExportValid,
-				apisv1alpha1.InternalErrorReason,
+				apisv1alpha1.APIExportEndpointSliceURLsReady,
+				apisv1alpha1.ErrorGeneratingURLsReason,
 				conditionsv1alpha1.ConditionSeverityError,
-				"Error getting APIExport %s|%s",
-				apiExportPath,
-				apiExportEndpointSlice.Spec.APIExport.Name,
+				"",
```

Comment:
should we also add the error here ? to indicate why it is invalid?

Reply:
as per the above. It is to avoid leaking information.